src/plugin/featureaction/featuresoundaction.js

```diff
@@ -0,0 +1,193 @@
+goog.require('os.im.action.AbstractImportAction');
+goog.require('os.implements');
+goog.require('os.legend');
+goog.require('os.legend.ILegendRenderer');
+goog.require('os.object');
+goog.require('os.style');
+goog.require('os.xml');
+
+/**
+ * Tag names used for XML persistence.
+ * @enum {string}
+ */
+plugin.im.action.feature.SoundActionTagName = {
+  SOUND: 'sound'
+};
+
+/**
+ * Import action that sets the sound for a {@link ol.Feature}.
+ * @extends {os.im.action.AbstractImportAction<ol.Feature>}
+ * @implements {os.legend.ILegendRenderer}
+ * @constructor
+ */
+plugin.im.action.feature.SoundAction = function() {
+  plugin.im.action.feature.SoundAction.base(this, 'constructor');
+
+  this.id = plugin.im.action.feature.SoundAction.ID;
+  this.label = plugin.im.action.feature.SoundAction.LABEL;
+  this.xmlType = plugin.im.action.feature.SoundAction.ID;
+  this.configUI = plugin.im.action.feature.SoundAction.CONFIG_UI;
+
+
+  /**
+   * Unique identifier for this action.
+   * @type {number}
+   * @protected
+   */
+  this.uid = goog.getUid(this);
+
+  /**
+   * The feature sound config.
+   * @type {!Object}
+   */
+  this.soundConfig = /** @type {!Object} */ (os.object.unsafeClone(
+      plugin.im.action.feature.SoundAction.DEFAULT_CONFIG));
+};
+goog.inherits(plugin.im.action.feature.SoundAction,
+    os.im.action.AbstractImportAction);
+
+os.implements(plugin.im.action.feature.SoundAction,
+    os.legend.ILegendRenderer.ID);
+
+/**
+ * Action identifier.
+ * @type {string}
+ * @const
+ */
+plugin.im.action.feature.SoundAction.ID = 'featureSoundAction';
+
+/**
+ * Property set on features to indicate they're using a feature sound action.
+ * @type {string}
+ * @const
+ */
+plugin.im.action.feature.SoundAction.FEATURE_ID = '_featureSoundAction';
+
+/**
+ * Action Label.
+ * @type {string}
+ * @const
+ */
+plugin.im.action.feature.SoundAction.LABEL = 'Set Sound';
+
+/**
+ * Action edit UI.
+ * @type {string}
+ * @const
+ */
+plugin.im.action.feature.SoundAction.CONFIG_UI = 'featureactionsoundconfig';
+
+/**
+ * The default sound configuration.
+ * @type {!Object}
+ * @const
+ */
+plugin.im.action.feature.SoundAction.DEFAULT_CONFIG = {
+  'sound': 'Default'
+
+};
+
+/**
+ * @inheritDoc
+ */
+plugin.im.action.feature.SoundAction.prototype.execute = function(items) {
+  for (var i = 0; i < items.length; i++) {
+    var item = items[i];
+    if (item) {
+      var featureConfig = /** @type {Array|Object|undefined} */ (item.get(
+          os.style.StyleType.FEATURE)) || {};
+      if (goog.isArray(featureConfig)) {
+        for (var j = 0; j < featureConfig.length; j++) {
+          os.style.mergeConfig(this.soundConfig, featureConfig[j]);
+        }
+      } else {
+        os.style.mergeConfig(this.soundConfig, featureConfig);
+      }
+
+      item.set(os.style.StyleType.FEATURE, featureConfig, true);
+      item.set(plugin.im.action.feature.SoundAction.FEATURE_ID, this.uid, true);
+
+      var configSound = this.soundConfig['sound'];
```

Comment:
Is this property read anywhere? If the intent is to play a sound when a feature matching this filter is loaded, then you don't need to save the sound to the feature.

Reply:
So the idea of this ticket is to play the sound when creating a feature action with a filter (ex ALTITIUDE greater than 0 ) Unfortunately I dont have anyway to stream features in the unclass environment. I'm not sure exactly how that works and I wasnt sure if it needed to be saved to the feature or not. But I removed it since it doesnt have a purpose to save it to the feature from my side.